api/wasm.go

```diff
@@ -162,6 +146,30 @@ type Module interface {
 
 	// ExportedGlobal a global exported from this module or nil if it wasn't.
 	ExportedGlobal(name string) Global
+
+	// ModuleCloser closes this module.
+	ModuleCloser
+}
+
+// ModuleCloser closes a module with an exit code.
+//
+// Note: This is an interface for decoupling, not third-party implementations. All implementations are in wazero.
+type ModuleCloser interface {
+	// Close is a convenience that invokes CloseWithExitCode with zero.
+	// Note: When the context is nil, it defaults to context.Background.
+	Close(context.Context) error
+
+	// CloseWithExitCode releases resources allocated for a Module. Use a non-zero exitCode parameter to indicate a
```

Comment:
the main clarification when this is Closer not ModuleCloser is while `Runtime.CloseXXX` closes other things (including code under a module), when it is closing a module those will get this exit code.

Reply:
Yeah forgot we're closing more than modules here.

Reply:
Tweaked the doc to try to point out it can release many things, but when they're Modules they'll get the exit code